From 8c730c58db75d064a664c2cccf50978e3cf330de Mon Sep 17 00:00:00 2001
From: betacentauri <@>
Date: Wed, 19 Jun 2013 22:05:45 +0200
Subject: [PATCH] Fix for flac files with high sample rates

Playback of flac files with high sample rates (e.g. 96000Hz) don't
work properly, because driver or hardware seems not to support them.
Limit gstreamer dvb audio sink sample rate capability to 48000Hz fixes
the problem.
---
 gstdvbaudiosink.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gstdvbaudiosink.c b/gstdvbaudiosink.c
index bec57f4..126c079 100644
--- a/gstdvbaudiosink.c
+++ b/gstdvbaudiosink.c
@@ -134,24 +134,24 @@ static guint gst_dvbaudiosink_signals[LAST_SIGNAL] = { 0 };
 		"signed = (boolean) { TRUE, FALSE }, " \
 		"width = (int) 32, " \
 		"depth = (int) 32, " \
-		"rate = (int) [ 1, MAX ], " "channels = (int) [ 1, 2 ]; " \
+		"rate = (int) [ 1, 48000 ], " "channels = (int) [ 1, 2 ]; " \
 		"audio/x-raw-int, " \
 		"endianness = (int) { " G_STRINGIFY(G_BYTE_ORDER) " }, " \
 		"signed = (boolean) { TRUE, FALSE }, " \
 		"width = (int) 24, " \
 		"depth = (int) 24, " \
-		"rate = (int) [ 1, MAX ], " "channels = (int) [ 1, 2 ]; " \
+		"rate = (int) [ 1, 48000 ], " "channels = (int) [ 1, 2 ]; " \
 		"audio/x-raw-int, " \
 		"endianness = (int) { " G_STRINGIFY(G_BYTE_ORDER) " }, " \
 		"signed = (boolean) { TRUE, FALSE }, " \
 		"width = (int) 16, " \
 		"depth = (int) 16, " \
-		"rate = (int) [ 1, MAX ], " "channels = (int) [ 1, 2 ]; " \
+		"rate = (int) [ 1, 48000 ], " "channels = (int) [ 1, 2 ]; " \
 		"audio/x-raw-int, " \
 		"signed = (boolean) { TRUE, FALSE }, " \
 		"width = (int) 8, " \
 		"depth = (int) 8, " \
-		"rate = (int) [ 1, MAX ], " "channels = (int) [ 1, 2 ];"
+		"rate = (int) [ 1, 48000 ], " "channels = (int) [ 1, 2 ];"
 
 static GstStaticPadTemplate sink_factory =
 GST_STATIC_PAD_TEMPLATE(
-- 
1.7.9.5

